### ISSUE TITLE
1.16.20.3 doesn't start on MacOS Catalina (ng)

### ISSUE THREAD

**Issue by andy-mortimer** [ChristopherHX/mcpelauncher-manifest#130]
Now there's a new release and my realm has been updated, I am motivated to get it running!

Previously I was running .168 because of the existing problems https://github.com/minecraft-linux/libc-shim/issues/3 with .175 on Catalina, which https://github.com/minecraft-linux/libc-shim/pull/4 fixes for me, although I admit the fix doesn't make a lot of sense.

With that fix in place, I hit a missing symbol error for sincosf, fixed by https://github.com/minecraft-linux/libc-shim/pull/5 .

With both of those fixes made, I am stuck, with the following error.

```
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Failed to find gl::supportsImmediateMode
```

I took a look at the code which produces this error and I am way out of my depth! Happy to take suggestions / pointers, although it may be a few days before I can look at this again.

This issue may be a duplicate of https://github.com/ChristopherHX/mcpelauncher-manifest/issues/129 , feel free to close it as such if it is.

**Comment by ChristopherHX** [comment]
You can get it running without xbox live, no realms access.

"gl::supportsImmediateMode"
change this to false for mac https://github.com/minecraft-linux/mcpelauncher-client/blob/45ed4f70094f3e0b6d0b386ce3e54a63b29912e2/src/gl_core_patch.cpp#L94

prepend this before the command
`LD_LIBRARY_PATH=/PATH/TO/FOLDER/OF/LIBEGL.DYLIB`
this file can be obtained by looking into the Google Chrome app or by compiling "google angle"

**Comment by andy-mortimer** [comment]
Thanks @ChristopherHX ! Confirmed, after switching that flag to false and putting libEGL.dylib on the LD_LIBRARY_PATH, mcpelauncher starts fine and loads/plays a local save. That's encouraging to see.

If I didn't want to use Realms, though, I would simply stick with 1.16.10.2!

Let me know if there's anything more I can do to help investigate/debug/fix this error.

**Comment by ChristopherHX** [comment]
Skilled people can try latest buggy stuff (1.16.20.03 x86_64 and Xboxlive works)
https://github.com/minecraft-linux/mcpelauncher-manifest/issues/374#issuecomment-674563699

No idea how hard it is to try this on macOS..

**Comment by ChristopherHX** [comment]
macOS works now linked message / feature-xal updated.
It crashed for me 5/6 tries to open, but stabilize this xbox shit is not my hobby.

**Comment by Aaronlei17** [comment]
> macOS works now linked message / feature-xal updated.
> It crashed for me 5/6 tries to open, but stabilize this xbox shit is not my hobby.

Does it work on 1.16.2.03 on macOS Catalina 1.15.5?

**Comment by chiefy** [comment]
@ChristopherHX thanks so much for your work. Anyhow, I am using the latest OSX build (190) and getting the dreaded 

```
libc++abi: terminating with uncaught exception of type std::runtime_error: Unknown socket family when converting sockaddr from host
```

```
Darwin XXXX 18.6.0 Darwin Kernel Version 18.6.0: Thu Apr 25 23:16:27 PDT 2019; root:xnu-4903.261.4~2/RELEASE_X86_64 x86_64
```

`10.14.5`

**Comment by ChristopherHX** [comment]
@chiefy Yes please help fixing this issue, not reproducible on my machine.
Only some research has been already done, about changes needed for macOS.
https://github.com/minecraft-linux/libc-shim/issues/3

**Comment by ChristopherHX** [comment]
@chiefy @andy-mortimer 
If you have time please tell me if I was able to fix the invalid socket addr in https://github.com/ChristopherHX/osx-packaging-scripts/releases/download/ng.dmg/Minecraft_Bedrock_Launcher_0.0.192_macOS_10.13.0.dmg
You should be able to run and login with 1.16.40.02

**Comment by andy-mortimer** [comment]
@ChristopherHX awesome! The linked .dmg starts fine, on the Catalina machine where it was failing before, and the login worked well too. I was able to access my Bedrock realm for the first time since the new version was released. Thank you so much for all the time you put into this!

**Comment by andy-mortimer** [comment]
@ChristopherHX it doesn't seem to be running very well, though. The launcher process and the game process are both using 100% CPU, and even on the title screen the game freezes for a couple of seconds, runs for a couple of seconds, freezes for a couple of seconds, and so on. Let me know if you need more detail.

**Comment by ChristopherHX** [comment]
The problem of 1.16.20-99 is missing symbols and or a magic pattern from mrarm to render with opengl, because it normally uses opengl es2 which is unavailable on mac so higher cpu usage till 1.16.100.

**Comment by ChristopherHX** [comment]
For the launcher process please toggle
disable gamelog to make the aggresive logging less impacting cpu usage

**Comment by bli488** [comment]
@ChristopherHX  for some reason the new launcher does not work for any of the versions because it says that i don’t have ssse cpu support what do i do

**Comment by ChristopherHX** [comment]
Does the old launcher even work on your PC? if yes where you get it.
Not having SSSE3 normally crashs during loading the game so I added a gui message. (currently blocks if cpu doesn't report support for SSSE3,...)

**Comment by bli488** [comment]
I have a Catalina at 10.15.6/the latest version, this is a brand new pc i got it last month, and the old launcher still works(0.1.168).

**Comment by bli488** [comment]
I got the old launcher from https://mcpelauncher.readthedocs.io/en/latest/

**Comment by ChristopherHX** [comment]
@bli488 Dmg filename of the not working version?
0.x.192 was published nth times for me to test.
ng 0.x.168 does the ssse3 check the same way, it should tell you x86_32 is unavailable in catalina, what is the compatibility report of the app as screenshot or text? Which cpu model?
My last test release published to github is
https://github.com/ChristopherHX/osx-packaging-scripts/releases/download/ng.dmg/Minecraft_Bedrock_Launcher_0.0.209_macOS_10.13.0.dmg
0.*.168 latest general available via updater

**Comment by bli488** [comment]
@ChristopherHX Thank you so much for the newest version it actually works 1/4 times, on the version i was using recently all it said in that was an exit code 6 it is the same in this version but it doesn’t  block you entirely so overall im happy thank you for your hard work

**Comment by nedbert** [comment]
Thanks @ChristopherHX I just download the link you posted and it works on Catalina 10.15.6.  Occasionally it will crash or hang when closing but all-in-all it seems to be working great.  I really appreciate the time you invest to keep this project running.

**Comment by ChristopherHX** [comment]
Does the current version v0.1-p4 work for more than 1/4 of times launching the game?
Or this issue stays alive.
Back then my machine wasn't an reliable enough to consider it as fixed.

**Comment by ChristopherHX** [comment]
Also please test beta 1.16.100.59 beta.
Sound of i386 (32bit) versions is dropped in 1.16.20+ macOS.

**Comment by nedbert** [comment]
This version works for me almost 99% of the time ... occasionally it crashes but not very often.   .... 0.1.p4-macOS-x86_64-0.2.243

**Comment by nedbert** [comment]
Hi @ChristopherHX ... been pretty busy as of late ... how do I test the beta?  Is it a Minecraft apk or a version of the launcher?

**Comment by ChristopherHX** [comment]
You find them hidden in the launcher, to prevent new people to accidentally play betas and post useless crashs.
Some of them are in the versionslist of the profile editor, after you have signed up for minecraft beta (play store) and enabled "show beta" or "show incompatible versions" in launcher setting, please turn both off if you want to play normal
_Please note latest beta's 1.16.100.60 - 1.16.200.52 won't work with that release._

**Comment by Kas-tle** [comment]
Interestingly 1.16.100.60 works with those versions for me (MacOS 10.14), minus sound as you stated would not work earlier. Sound on 1.16.20.3, however, is still just fine. Seems to have worked pretty consistently though. I haven't experienced any crashes that seem to be related to launcher instability. The 200 versions are still a no-go, of course. This is for v0.1-p4 build 243.

**Comment by nedbert** [comment]
Hi @ChristopherHX ... thanks again for creating this launcher:

Is there a different issue I should watch ... that has been reported for the latest version (1.16.210.05) posted on the Google Play store?  It shows as unsupported in my launcher.  It was odd because my launcher pulled an update this morning and I thought it might contain support for 1.16.210.05.   ;-)

The launcher has been working great for me for many months now THANK YOU for that !!

Let me know if I can help test in any way.

I have Version 0.1.b5-macOS-x86_64-0.2.280 (build 280)

12:22:19 Trace [Launcher] Loading Minecraft library
12:22:20 Error [MinecraftUtils] Failed to load Minecraft: dlopen failed: cannot locate symbol "malloc_usable_size" referenced by "/Users/nedbauerle/Library/Application Support/mcpelauncher/versions/1.16.210.05/lib/x86_64/libminecraftpe.so"...
12:22:20 Error [Launcher] Failed to load Minecraft library, please reinstall or wait for an update to support the new release

Incompatible Minecraft installation, please select a different or older Version
This Launcher is a free Open Source Software which usually fell behind official updates from Google Play
In some cases there are missing game files,
missing Symbols expected to be provided by this Launcher via updates
 or otherwise broke the Launcher


<img width="640" alt="Screen Shot 2021-03-10 at 12 23 48 PM" src="https://user-images.githubusercontent.com/1257735/110671352-6556ca00-819c-11eb-9d26-0f83a527d410.png">

**Comment by ChristopherHX** [comment]
Version 0.1.b5-macOS-x86_64-0.2.280 (build 280) only fix this https://github.com/ChristopherHX/mcpelauncher-manifest/issues/168

Someone already want to make a PR (https://github.com/minecraft-linux/libc-shim/pull/7)

Please subscribe this issue https://github.com/ChristopherHX/mcpelauncher-manifest/issues/48#issuecomment-795472828 to get updates about the version support
